modules/private/internal.go

```diff
@@ -41,8 +43,34 @@ func newInternalRequest(url, method string) *httplib.Request {
 	})
 	if setting.Protocol == setting.UnixSocket {
 		req.SetTransport(&http.Transport{
-			Dial: func(_, _ string) (net.Conn, error) {
-				return net.Dial("unix", setting.HTTPAddr)
+			DialContext: func(ctx context.Context, _, _ string) (net.Conn, error) {
+				var d net.Dialer
+				conn, err := d.DialContext(ctx, "unix", setting.HTTPAddr)
+				if err != nil {
+					return conn, err
+				}
+				if setting.LocalUseProxyProtocol {
```

Comment:
In which case we need this?

Reply:
The PROXY protocol is very strict. The PROXY header must be there. If you set your local root url to be talking to gitea direct instead of through the proxy then you have to send the PROXY header - hence this setting. If however, you set your local root url to direct through the proxy then you do not need this header.

It defaults to the value of USE_PROXY_PROTOCOL but it's a good idea to provide an option to switch it off if necessary.